lib/source/defender/aws_iot_defender_api.c

```diff
@@ -237,57 +271,58 @@ void AwsIotDefender_Stop( void )
     if( !_started )
     {
         IotLogWarn( "Defender has not started yet." );
-
-        return;
     }
-
-    /* Wait for all the metrics processing to be done, if there are outstanding requests */
-    if( IotSemaphore_GetCount( &_doneSem ) > MAX_DEFENDER_OUTSTANDING_PUBLISH_REQ )
+    else
     {
+        /* Wait for all the metrics processing to be done, if there are outstanding requests */
         IotSemaphore_Wait( &_doneSem );
```

Comment:
Why is this a blocking call? Wouldn't this call be vulnerable of making the application thread go into an indefinite wait state?

Reply:
When application calls stop, library needs to wait till taskpool thread has finished executing publish job, otherwise connection will be destroyed while it is being used.